src/scala/scripts/TwitterScroogeGenerator.scala

```diff
@@ -0,0 +1,269 @@
+  override def visitFile(file: Path, attr: BasicFileAttributes) = {
+    Files.delete(file)
+    FileVisitResult.CONTINUE
+  }
+
+  override def postVisitDirectory(dir: Path, e: IOException) = {
+    if (e != null) throw e
+    Files.delete(dir)
+    FileVisitResult.CONTINUE
+  }
+}
+
+case class ForeachFile(f: Path => Unit) extends SimpleFileVisitor[Path] {
+  override def visitFile(file: Path, attr: BasicFileAttributes) = {
+    f(file)
+    FileVisitResult.CONTINUE
+  }
+}
+
+//TODO add logging?
+object ScroogeGenerator {
+  def deleteDir(path: Path) {
+    try {
+      Files.walkFileTree(path, DeleteRecursively)
+    } catch {
+      case e: Exception => //TODO LOG
+    }
+  }
+
+  def extractJarTo(_jar: String, _dest: String): List[Path] = {
+    val files = Buffer[Path]()
+    //TODO add to it, error if duplicate!
+    val jar = new JarFile(_jar)
+    val enumEntries = jar.entries()
+    while (enumEntries.hasMoreElements) {
+      val file = enumEntries.nextElement().asInstanceOf[JarEntry]
+      val f = new File(_dest, file.getName) //TODO pathify this
+      if (file.isDirectory()) f.mkdir()
+      else {
+        val path = Paths.get(f.toURI)
+
+        val is = jar.getInputStream(file)
+        try {
+          Files.copy(is, path) // Will error out if path already exists
+        } finally {
+          is.close()
+        }
+
+        files += path
+      }
+    }
+    files.toList
+  }
+
+  //TODO add logging that can be turned on/off?
+  def main(args: Array[String]) {
+    if (args.length < 4) sys.error("Need to ensure enough arguments! " +
+      "Required 3 arguments: onlyTransitiveThriftSrcs immediateThriftSrcs " +
+      "jarOutput remoteJarsFile. Received: " + args.mkString(","))
+
+    val onlyTransitiveThriftSrcsFile = args(0)
+    val immediateThriftSrcsFile = args(1)
+    val jarOutput = args(2)
+    val remoteJarsFile = args(3)
+
+    val tmp = Paths.get(Option(System.getenv("TMPDIR")).getOrElse("/tmp"))
+    val scroogeOutput = Files.createTempDirectory(tmp, "scrooge").toString
+
+    // These are all of the files to include when generating scrooge
+    // Should not include anything in immediateThriftSrcs
+    val onlyTransitiveThriftSrcJars =
+      Source.fromFile(onlyTransitiveThriftSrcsFile).getLines.toSet
+
+    // These are the files whose output we want
+    val immediateThriftSrcJars =
+      Source.fromFile(immediateThriftSrcsFile).getLines.toSet
+
+    val genFileMap = s"$scroogeOutput/gen-file-map.txt"
+
+    val scrooge = new Compiler
+
+    // we need to extract into the same tree, as that is the only way to get relative imports between them working..
+    // AS SUCH, we are just going to try extracting EVERYTHING to the same tree, and we will just error if there
+    // are more than one.
+    val _tmp =  Files.createTempDirectory(tmp, "jar")
+    // This will only be meaningful if they have absolute_prefix set
+    scrooge.includePaths += _tmp.toString
```

Comment:
actually, includePaths accepts jars:

https://github.com/twitter/scrooge/blob/develop/scrooge-generator/src/main/scala/com/twitter/scrooge/frontend/Importer.scala#L56

https://github.com/twitter/scrooge/blob/develop/scrooge-generator/src/main/scala/com/twitter/scrooge/Compiler.scala#L64

so we can just point to a zip or a jar.